src/OrchardCore.Modules/OrchardCore.Forms/Workflows/Activities/ValidateRuleFormTask.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Localization;
+using Newtonsoft.Json;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.Forms.Extensions;
+using OrchardCore.Forms.Models;
+using OrchardCore.Workflows.Abstractions.Models;
+using OrchardCore.Workflows.Activities;
+using OrchardCore.Workflows.Models;
+
+namespace OrchardCore.Forms.Workflows.Activities
+{
+    public class ValidateRuleFormTask : TaskActivity
+    {
+        private readonly IUpdateModelAccessor _updateModelAccessor;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IStringLocalizer S;
+
+        public ValidateRuleFormTask(
+            IHttpContextAccessor httpContextAccessor,
+            IUpdateModelAccessor updateModelAccessor,
+            IStringLocalizer<ValidateRuleFormTask> localizer
+        )
+        {
+            _updateModelAccessor = updateModelAccessor;
+            _httpContextAccessor = httpContextAccessor;
+            S = localizer;
+        }
+
+        public override string Name => nameof(ValidateRuleFormTask);
+
+        public override LocalizedString DisplayText => S["Validate Rule Form Task"];
+
+        public override LocalizedString Category => S["Validation"];
+
+        public override bool HasEditor => false;
+
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(S["Valid"], S["Invalid"]);
+        }
+
+        public override ActivityExecutionResult Execute(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var updater = _updateModelAccessor.ModelUpdater;
+
+            if (updater == null)
+            {
+                throw new InvalidOperationException("Cannot add model validation errors when there's no Updater present.");
+            }
+
+            var httpContext = _httpContextAccessor.HttpContext;
+            var validateRule = true;
+            var rules = new List<ValidationRule>();
+            foreach (var item in httpContext.Request.Form)
+            {
+                if (item.Key.Equals("validationRules", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    if (!String.IsNullOrEmpty(item.Value[0]))
+                    {
+                        rules = JsonConvert.DeserializeObject<List<ValidationRule>>(item.Value[0]);
```

Comment:
This fails the whole reason of the server-side validation. Anyone can hack the payload with custom validation that will invalidate any validation.